### PR TITLE
o/ifacestate: add convenience Active() method to ConnectionState struct

### PR DESCRIPTION
**Pull request by stolowski**
There are a couple of places where a ConnectionState is checked for hotplug gone and undesired flag (upcoming #10864 is another one), so I though of reducing it by a tiny bit by adding `Active()` convenience method.

My initial idea was to have a helper for iterating over conns with a lambda for the actual logic (because this is the pattern for those places), but there is no good way of doing this without making it more complex, so it's not a win.

I'll update this PR after #10864 lands.

### CODE REVIEW

**Review by MiguelPires — APPROVED** [review]
LGTM

**Review by mvo5 — APPROVED** [review]
This looks fine to me, any reason to keep it in draft?

**Comment by stolowski** [comment]
> This looks fine to me, any reason to keep it in draft?

Yes, I'm waiting for @mardy 's mount control PR #3 to land and will then simplify one more line of his new code.

**Comment by stolowski** [comment]
@mvo5 I've updated it after Alberto's change landed - one new commit. It's not draft anymore.

**Review by mardy — APPROVED** [review]
LGTM!

**Pull request merged**
